packages/client/src/runtime/core/model/aggregates/groupBy.ts

```diff
@@ -10,13 +11,15 @@ import { desugarUserArgs as desugarUserArgsAggregate } from './aggregate'
  * @param userArgs to transform
  * @returns
  */
-function desugarUserArgs(userArgs: object) {
+function desugarUserArgs(userArgs: UserArgs) {
   const _userArgs = desugarUserArgsAggregate(userArgs)
 
   // we desugar the array into { [key]: boolean }
-  if (Array.isArray(userArgs?.['by']) === true) {
-    for (const key of userArgs?.['by'] ?? []) {
-      _userArgs['select'][key] = true
+  if (Array.isArray(userArgs['by'])) {
+    for (const key of userArgs['by']) {
+      if (_userArgs['select'] && typeof key === 'string') {
```

Comment:
Is the first check here to make the type checker happy? As far as I understand, `_userArgs['select']` is always present since it's the result of `desugarUserArgsAggregate`.

Reply:
Yep, probably overkill, I adjusted the return type in the parent call instead.